plugins/woocommerce-admin/client/products/sections/images-section.tsx

```diff
@@ -174,19 +175,22 @@ export const ImagesSection: React.FC = () => {
 						) : (
 							<CardBody>
 								<MediaUploader
+									multipleSelect={ true }
 									onError={ () => null }
-									onSelect={ ( file ) => {
+									onSelect={ ( files ) => {
 										if (
-											images.find(
-												( img ) => file.id === img.id
-											) === undefined
+											intersectionBy(
+												images,
+												files,
+												'id'
+											).length === 0
```

Comment:
This prevents any image from being added as soon as one duplicate image is part of the list.
Could we just filter out the duplicate images? and add all the newly selected images?

Reply:
Good idea! I fixed that in the commit [9a29f9c](https://github.com/woocommerce/woocommerce/pull/35722/commits/9a29f9ce1d073ccfaa8bd78702ab562b80e77411)